cfpb/consumerfinance.gov
# Mega Menu: add tabindex and aria behavior for tabbing/voiceover of off-screen content on mobile

It was possible to tab to and voiceover offscreen menu content. This PR sets the `tabindex` and `aria-hidden` attributes on the 1st level links when they are off-screen.

## Changes

- Add viewport sizes to mega menu tests.
- Add basic test for presence/absence of `tabindex` when navigating the menu.
- Add logic to enable/disable 1st level menu items when it moves on/off screen.


## How to test this PR

1. Pull branch and run `gulp clean && gulp build`.
2. Visit any page and resize to mobile screen size.
3. Tab till the hamburger menu is focused. Tab again you'll be on the tab trigger (separate issue to remove this some time). Tab a third time and you should be `/coronavirus/` link. You can log `document.activeElement` in the console if it's unclear what has focus. Shift+tab to move back over the hamburger menu. Compare to production where tabbing enters the off-screen links within the menu.
4. Open the menu and tab through it and once it reaches the end it should close.
5. Open the menu and click (or tab and spacebar or return key) the "News" item to go to the 2nd level menu.
6. Tab through the 2nd level menu. Once reaching the end the menu should close. Compare to production where the menu first tabs through the off-screen 1st level menu items before closing.
7. Open the menu and click on any item to go to the 2nd level menu. Click the semi-transparent overlay to close the menu. Tab through the page like step 3 to see that not possible to tab into the off-screen 1st level menu items.
8. Open the search and repeat the above.
9. Resize to desktop and back to mobile and repeat the above.
10. Repeat all with Voiceover and ensure you off-screen content isn't read.
